File: dart_lang/lib/behavioral/visitor/day_shift_schedule_management.dart
import 'package:Strategy_design/behavioral/visitor/schedule_magngement.dart';
import 'package:Strategy_design/behavioral/visitor/schedule_management_visitor.dart';

class DayShiftScheduleManagement implements ScheduleManagement {
  @override
  void accept(ScheduleManagementVisitor scheduleManagementVisitor) {
    scheduleManagementVisitor.visitDayShift(this);
  }

  @override
  void calcOverTime() {
    print(' Calculating over time for day shift employees');
  }

  @override
  void generateReport() {
    print('Generating report for day shift employees');
  }
}

File: typescript/patterns/visitor/day-shift-schedule-management.ts
class DayShiftScheduleManagement implements ScheduleManagement {
  generateReport() {
    console.log("Generating report for day shift...");
  }

  calculateOverTime() {
    console.log("Calculating overtime for day shift...");
  }

  accept(visitor: ScheduleManagementVisitor): void {
    visitor.visit(this, "day");
  }
}
